### PR TITLE
Bump to v4.2.1

### PR DESCRIPTION
**Pull request by mheon**
Final backports and release notes for v4.2.1, plus the actual release bump.

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,5 +1,37 @@
 # Release Notes
 
+## 4.2.1
+### Features
+- Added support for Sigstore signatures (`sigstoreSigned`) to the `podman image trust set` and `podman image trust show` commands.`
+- The `podman image trust show` command now recognizes new `lookaside` field names.
+- The `podman image trust show` command now recognizes `keyPaths` in `signedBy` entries.
+
+### Changes
+- BREAKING CHANGE: `podman image trust show` may now show multiple entries for the same scope, to better represent separate requirements. GPG IDs on a single row now always represent alternative keys, only one of which is required; if multiple sets of keys are required, each is represented by a single line.
```
Comment (Luap99) [inline RELEASE_NOTES.md:10]: May I question why a patch version bump contains a breaking changes?

Comment (mheon) [reply RELEASE_NOTES.md:10]: We had to throw the same in RHEL's 4.2.0, so we put it here as well to keep the branches as close to in sync as we can. Possibly not the best idea, but we didn't really have a good option.

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@vrothberg @containers/podman-maintainers PTAL

**Review by vrothberg — COMMENTED** [review]
/lgtm

**Pull request merged**